packages/dashboard/src/pages/Settings/components/dialogs/RestoreDialog.tsx

```diff
@@ -0,0 +1,165 @@
+import { useState } from 'react'
+import ConfirmDialog from '../../../../components/ConfirmDialog'
+import { MaskColorVar } from '@dimensiondev/maskbook-theme'
+import FileUpload from '../../../../components/FileUpload'
+import { useAsync } from 'react-use'
+import { Services } from '../../../../API'
+import BackupPreviewCard from '../BackupPreviewCard'
+
+const useStyles = makeStyles(() => ({
+    container: { flex: 1 },
+    hide: { display: 'none' },
+}))
+
+const StyledTabList: typeof TabList = styled(TabList)(() => ({
+    [`& .${tabsClasses.indicator}`]: {
+        display: 'none',
+    },
+})) as any
+
+const StyledTab: typeof Tab = styled(Tab)(() => ({
+    [`&.${tabClasses.root}`]: {
+        flex: 1,
+        background: MaskColorVar.secondaryBackground,
+        textTransform: 'none',
+        [`&:first-of-type`]: {
+            borderTopLeftRadius: 4,
+            borderBottomLeftRadius: 4,
+        },
+        [`&:last-of-type`]: {
+            borderTopRightRadius: 4,
+            borderBottomRightRadius: 4,
+        },
+        [`&.${tabClasses.selected}`]: {
+            color: MaskColorVar.primaryContrastText,
+            background: MaskColorVar.primary,
+        },
+    },
+})) as any
+
+const StyledTabPanel = styled(TabPanel)(({ theme }) => ({
+    [`&.${tabPanelClasses.root}`]: {
+        marginTop: theme.spacing(2.5),
+        padding: 0,
+    },
+}))
+
+const TextArea = styled(InputBase)(({ theme }) => ({
+    [`&.${inputBaseClasses.root}`]: {
+        height: 180,
+        alignItems: 'flex-start',
+        background: MaskColorVar.secondaryBackground,
+        borderRadius: 8,
+        padding: theme.spacing(2),
+    },
+}))
+
+interface RestoreDialogProps {
+    open: boolean
+    onClose(): void
+}
+
+export default function RestoreDialog({ open, onClose }: RestoreDialogProps) {
+    const classes = useStyles()
+    // tab switch
+    const [tab, setTab] = useState('file')
+    // paste text
+    const [text, setText] = useState('')
+    // file content
+    const [content, setContent] = useState('')
+    // parsed json
+    const [json, setJSON] = useState<any>(null)
+
+    const handleClose = () => {
+        onClose()
+        setTab('file')
+        setText('')
+        setContent('')
+        setJSON(null)
+    }
+    const handleConfirm = async () => {
+        if (!json) return
+
+        const permissions = permission.value ?? []
+
+        if (permissions.length) {
+            const granted = await Services.Welcome.requestPermissions(permissions)
+            if (!granted) return
+        }
+
+        await Services.Welcome.restoreBackup(json)
+    }
+    const handleTabChange = (event: React.SyntheticEvent, newValue: string) => {
+        setTab(newValue)
+    }
+    const handleFileChange = (file: File, content?: string) => {
+        setContent(content || '')
+    }
+    const handleTextChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setText(event.target.value)
+    }
```

Comment:
If you inline those functions, you can get rid of the typing on the parameter.

Reply:
changed to inline functions